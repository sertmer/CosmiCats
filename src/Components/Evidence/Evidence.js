import React from 'react';
import './Evidence.scss';

export const Evidence = ({evidence}) => {
  return (
    <article className='evidence-card'>
      <img className='evidence-img' src={evidence.img}/>
      <div className='evidence-details'>
        <p>{evidence.id}</p>
        <p>{evidence.dateCreated}</p>
        <button type='button'>delete</button>
      </div>
    </article>
  )
}

export default Evidence;