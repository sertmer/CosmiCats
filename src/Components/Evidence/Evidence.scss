@import '../../_scss/variables';
@import '../../_scss/mixins';

.evidence-card {
  width: 29.2%;
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
  padding: 1%;
  @include rounded-corners(5px);
  margin: 1%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.evidence-img {
  height: 400px;
  width: 400px;
  @include rounded-corners(5px);
}

.evidence-details {
  text-align: center;
}