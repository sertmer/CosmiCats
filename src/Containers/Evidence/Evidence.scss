@import '../../_scss/variables';
@import '../../_scss/mixins';

.evidence-card {
  width: 29.2%;
  background: linear-gradient(
    0deg, 
    rgba(193,193,193,1) 0%, 
    rgba(245,245,245,1) 100%
  );
  padding: 1%;
  @include rounded-corners(5px);
  margin: 1%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.evidence-img {
  height: 400px;
  width: 400px;
  @include rounded-corners(5px);
}

.evidence-details {
  text-align: center;
}

.delete-evidence-btn {
  width: 10em;
  @include rounded-corners(5px);
  background: rgb(69,198,141);
  background: linear-gradient(
    0deg, rgba(69,198,141,1) 0%, 
    rgba(73,210,149,1) 35%, 
    rgba(85,246,175,1) 100%
  );
  padding: 3%;
  @include rounded-corners(5px);
  font-size: 2em;
  font-family: $main-font;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.delete-evidence-btn:hover {
  transform: scale(1.1);
}