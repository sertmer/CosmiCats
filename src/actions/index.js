export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const setNasaImages = images => ({
  type: 'SET_NASA_IMAGES',
  images
})

export const setCatImages = images => ({
  type: 'SET_CAT_IMAGES',
  images
})

export const addToEvidence = evidence => ({
  type: 'ADD_EVIDENCE',
  evidence
})