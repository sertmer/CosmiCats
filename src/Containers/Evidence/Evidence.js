import React from 'react';
import { connect } from 'react-redux';
import { removeFromEvidence } from '../../actions/';
import './Evidence.scss';
import PropTypes from 'prop-types';
import { EvidenceContainer } from '../EvidenceContainer/EvidenceContainer';

export const Evidence = ({evidence, removeFromEvidence}) => {
  return (
    <article className='evidence-card'>
      <img className='evidence-img' src={evidence.img}/>
      <div className='evidence-details'>
        <h2>Nasa Identification Number</h2>
        <p>{evidence.id}</p>
        <h2>Date Captured</h2>
        <p>{evidence.dateCreated}</p>
        <button 
          className='delete-evidence-btn'
          type='button'
          onClick={() => removeFromEvidence(evidence)}
        >delete</button>
      </div>
    </article>
  )
}

export const mapDispatchToProps = dispatch => ({
  removeFromEvidence: evidence => dispatch(removeFromEvidence(evidence))
})

export default connect(null, mapDispatchToProps)(Evidence);

Evidence.propTypes = {
  evidence: PropTypes.object,
  removeFromEvidence: PropTypes.func
}