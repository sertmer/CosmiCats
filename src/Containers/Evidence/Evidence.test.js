import React from 'react';
import { shallow } from 'enzyme';
import { removeFromEvidence } from '../../actions'
import { Evidence, mapDispatchToProps } from './Evidence';

describe('Evidence', () => {
  let wrapper
  let mockEvidence
  let mockRemoveFromEvidence

  beforeEach(() =>{
    mockRemoveFromEvidence = jest.fn()
    mockEvidence = {
      img: 'someURL.com',
      id: 'ab1',
      dateCreated: '12-12-12'
    }

    wrapper = shallow(<Evidence 
      evidence={mockEvidence} 
      removeFromEvidence={mockRemoveFromEvidence}
    />)
  })

  describe('Evidence unit tests', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call removeFromEvidence when user clicks delete', () => {
      wrapper.find('button').simulate('click')

      expect(mockRemoveFromEvidence).toHaveBeenCalledWith(mockEvidence)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the removeEvidence action', () => {
      const mockDispatch = jest.fn()
      const mockRemoveFromEvidenceAction = removeFromEvidence(mockEvidence)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.removeFromEvidence(mockEvidence)

      expect(mockDispatch).toHaveBeenCalledWith(mockRemoveFromEvidenceAction)
    })
  })
})