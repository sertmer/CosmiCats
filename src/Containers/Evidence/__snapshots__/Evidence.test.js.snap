// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evidence Evidence unit tests should match the snapshot 1`] = `
<article
  className="evidence-card"
>
  <img
    className="evidence-img"
    src="someURL.com"
  />
  <div
    className="evidence-details"
  >
    <h2>
      Nasa Identification Number
    </h2>
    <p>
      ab1
    </p>
    <h2>
      Date Captured
    </h2>
    <p>
      12-12-12
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      delete
    </button>
  </div>
</article>
`;
